src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -2481,9 +2481,13 @@ public async Task<(long Index, object Result)> SendToLeaderAsync(CommandBase cmd
 
         //this is needed for cases where Result or any of its fields are blittable json.
         //(for example, this is needed for use with AddOrUpdateCompareExchangeCommand, since it returns BlittableJsonReaderObject as result)
-        public Task<(long Index, object Result)> SendToLeaderAsync(TransactionOperationContext context, CommandBase cmd)
+        public async Task<(long Index, object Result)> SendToLeaderAsync(TransactionOperationContext context, CommandBase cmd)
         {
-            return SendToLeaderAsyncInternal(context, cmd);
+            // the command and result here rely on the context, so we must prevent its early release.
+            using (context.PreventContextReturn())
```

Comment:
Why is this even thread safe in the first place?
When we initialize the call, we *know* no one else can be concurrent with us.
Then we just need to ensure that either:
* we are still waiting when `SendToLeaderAsyncInternal` returns
* we gave up, therefor we will move the ownership to `SendToLeaderAsyncInternal` return to clear it, no?

Reply:
My solution is not good, because we will not dispose the context, but return it still to the pool... which is bad.
The problem is that the `Value` of the compare exchange is owned by that context, so we must keep it alive.
Will try to clone the entire command to a different context...